bigchaindb/models.py

```diff
@@ -202,23 +202,28 @@ def validate(self, bigchain):
             OperationError: If a non-federation node signed the Block.
             InvalidSignature: If a Block's signature is invalid.
         """
+
+        self._validate_block(bigchain)
+        self._validate_block_transactions(bigchain)
+
+        return self
+
+    def _validate_block(self, bigchain):
```

Comment:
It might be worthwhile to add a comment to note that the hash (id) is validated in the `from_dict()` method, not here (in case someone wonders if/when that ever gets checked).